src/hash.c

```diff
@@ -8,12 +8,14 @@ struct hash_pair {
 };
 struct hash_tab {
   size_t size, free;
-  uintptr_t multiplier;
-  struct hash_pair tb[];
+  uintptr_t multiplier1, multiplier2;
+  struct hash_pair *table;  // Single table for double hashing
 };
 
 // TAOCP vol. 3, section 6.4: for multiplication hashing, use A ~ 1/phi, the golden ratio.
-static const double hash_multiplier = 0.618033988749895;
+// 
+static const double hash_multiplier1 = 0.618033988749895;
+static const double hash_multiplier2 = 0.316227766016838;
```

Comment:
Why make the second number specifically $\sqrt{0.1}$?

Reply:
No real deeper reasoning. In my algorithms course at university we learned that pi, e or sqrt(10) (as ancient pi approximation) are often used as hash multiplier. I guess in my head I checked finetuning hash multipliers as a future Todo.